lib/ipaddr.h

```diff
@@ -174,6 +174,21 @@ static inline bool ipaddr_is_same(const struct ipaddr *ip1,
 	return ipaddr_cmp(ip1, ip2) == 0;
 }
 
+static inline bool ipaddr_is_mcast(const struct ipaddr *ip)
```

Comment:
this first one looks like a grab-bag of 4 - 5 actual different commits.  I also really don't understand where we are going with IPADDR_NONE and how it's actually meant to be used.  More verbiage here would be good